omegaconf/nodes.py

```diff
@@ -218,6 +222,48 @@ def __deepcopy__(self, memo: Dict[int, Any]) -> "IntegerNode":
         return res
 
 
+class BytesNode(ValueNode):
+    def __init__(
+        self,
+        value: Any = None,
+        key: Any = None,
+        parent: Optional[Container] = None,
+        is_optional: bool = True,
+        flags: Optional[Dict[str, bool]] = None,
+    ):
+        super().__init__(
+            parent=parent,
+            value=value,
+            metadata=Metadata(
+                key=key,
+                optional=is_optional,
+                ref_type=bytes,
+                object_type=bytes,
+                flags=flags,
+            ),
+        )
+
+    def _validate_and_convert_impl(self, value: Any) -> bytes:
+        val: bytes
+        try:
+            if type(value) == bytes:
+                val = value
+            elif type(value) == str:
+                val = bytes.fromhex(value)
```

Comment:
Similar to the above comment on use of the `bytes.hex` method -- using `bytes(value, "utf-8")` could be an alternative here.

Reply:
maybe it would be better to use `encode` and `decode` ?

Reply:
Nice!